### PR TITLE
PWN-6262 - Off-ramp. Fetching items and showing in history

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-6262

## Description of Work

- reusable layout
- showing different items in history

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/sell/ui/extensions/SellTransactionDetailsBindingExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    isAddressSolWallet: Boolean,
+    walletAddress: String,
+    onAddressCopied: () -> Unit,
+) {
+    imageViewCopy.isVisible = isAddressSolWallet
+    imageViewCopy.setOnClickListener {
+        context.copyToClipBoard(walletAddress)
+        onAddressCopied()
+    }
+}
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/sell/ui/extensions/SellTransactionDetailsBindingExtensions.kt:181]: Imho, the UI rendering should be in fragment. 
It's not about static extensions.
The fragment which uses these extensions are absolutely empty, as if it has empty space for exactly this code

@kabramovp2p what do you think?

File: app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsBottomSheet.kt [hunk 2]
```diff
@@ -1,22 +1,65 @@
...
+    companion object {
+        fun show(fm: FragmentManager, details: SellTransactionDetails) {
+            SellTransactionDetailsBottomSheet()
+                .withArgs(ARG_DETAILS to details)
+                .show(fm, SelectTokenBottomSheet::javaClass.name)
+        }
+    }
 
-    override val presenter: SellTransactionDetailsContract.Presenter by inject()
+    override val presenter: SellTransactionDetailsContract.Presenter = SellTransactionDetailsPresenter()
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsBottomSheet.kt:38]: nice

File: app/src/main/java/org/p2p/wallet/history/ui/history/HistoryContract.kt [hunk 3]
```diff
@@ -3,15 +3,16 @@ package org.p2p.wallet.history.ui.history
...
 import org.p2p.wallet.history.model.HistoryTransaction
 
 interface HistoryContract {
     interface View : MvpView {
         fun showPagingState(state: PagingState)
-        fun showHistory(items: List<HistoryTransaction>)
         fun openTransactionDetailsScreen(transaction: HistoryTransaction)
         fun showRefreshing(isRefreshing: Boolean)
         fun scrollToTop()
+        fun showHistory(items: List<HistoryTransaction>, moonpayItems: List<HistoryItem.MoonpayTransactionItem>)
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/history/ui/history/HistoryContract.kt:15]: Why you didn't unify MoonpayTransaction to `HistoryTransaction`?

**Review by 1sakov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/sell/ui/extensions/SellTransactionDetailsBindingExtensions.kt (hunk 1 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/sell/ui/extensions/SellTransactionDetailsBindingExtensions.kt:181]: I agree to make UI related logic in fragment

File: app/src/main/java/org/p2p/wallet/history/ui/history/HistoryPresenter.kt [hunk 4]
```diff
@@ -81,15 +86,19 @@ class HistoryPresenter(
...
     private suspend fun fetchHistory(isRefresh: Boolean = false) {
         try {
+            val moonpayItems = sellInteractor.loadUserSellTransactionsDetails()
+                .map { HistoryItem.MoonpayTransactionItem(it) }
+            moonpayTransactions.clear()
+            moonpayTransactions.addAll(moonpayItems)
             val fetchedItems = historyInteractor.loadTransactions(isRefresh)
             transactions.addAll(fetchedItems)
-            view?.showHistory(transactions)
+            view?.showHistory(transactions, moonpayItems)
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/history/ui/history/HistoryPresenter.kt:95]: I would split the requests into separate suspend functions and call them here. 
This function is getting complex

**Review by kabramovp2p — APPROVED** [review]
do not like 2 transactions lists in adapter =(

File: app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsBottomSheet.kt (hunk 2 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsBottomSheet.kt:38]: thanks, going to fix that!

File: app/src/main/java/org/p2p/wallet/history/ui/history/HistoryContract.kt (hunk 3 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/history/ui/history/HistoryContract.kt:15]: HistoryTransaction has blockchain data inside, we don't have that in moonpay

File: app/src/main/java/org/p2p/wallet/history/ui/history/HistoryPresenter.kt (hunk 4 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/history/ui/history/HistoryPresenter.kt:95]: i'll do it in a separate task, history fragment is a mess that i will work on later

File: app/src/main/java/org/p2p/wallet/history/ui/history/HistoryContract.kt (hunk 3 above)
Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/history/ui/history/HistoryContract.kt:15]: I think we can remove default parameters from the sealed class and blockchain data will not be required then

File: app/src/main/java/org/p2p/wallet/sell/ui/extensions/SellTransactionDetailsBindingExtensions.kt (hunk 1 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/sell/ui/extensions/SellTransactionDetailsBindingExtensions.kt:181]: how can i use the same render logic in another fragment?

**Pull request closed without merging**